crates/pathfinder/src/state/sync.rs

```diff
@@ -961,6 +961,14 @@ async fn l2_update(
                 // error. It is unlikely that any error here wouldn't simply
                 // repeat indefinitely.
                 *websocket_txs = None;
+                return Ok(());
+            }
+            if sender.events.receiver_count() > 0 {
```

Comment:
Do we want to broadcast events from the pending block as well? That would complicate things a lot but users would get a much faster notification for emitted events.

Reply:
BTW, that would complicate things quite a lot, so even if we decide to implement that it might be worth splitting that into a new issue/PR.

Reply:
It will be a fair bit more complicated; but maybe that's fine?